src/sema/ast.rs

```diff
@@ -597,6 +597,13 @@ pub struct Contract {
     pub default_constructor: Option<(Function, usize)>,
     pub cfg: Vec<ControlFlowGraph>,
     pub code: Vec<u8>,
+    // This contract can be instantiated and called or receive funds.
+    // It implies that it has at least one of:
+    // * public messages
+    // * public storage (results in public getters)
+    // * fallback or receive function
+    // alternative names I came up: has_code, extrinsic, is_public, exterior, has_surface
+    pub has_callable_function: bool,
```

Comment:
`has_callable_function` is a substrate specific property. How do you feel about:
```rust
pub instantiable: bool,
```
By default this would have the value of `.is_concrete()`, i.e `matches!(ty, pt::ContractTy::Contract(_))` but the contract does not have a requisite function for Substrate, then it can be set to `false`. 

In the future, we could set this to `false` for any sema error and not generate code for this contract, rather than not generating code for all contracts if there is an error.

In various code paths where we use `.is_concrete()` we can replace this with `.instantiable` (or should that be `.is_instantiable`).

wdty? @LucasSte  @xermicus

Reply:
Mostly ACK. But need to be careful as `is_concrete()` is not the same as this bool flag. For this PR I'd be happy to change the name but I don't really fancy touching it in other places...